src/v/kafka/server/group.cc

```diff
@@ -119,8 +119,7 @@ static model::record_batch make_tx_batch(
   const model::producer_identity& pid,
   T cmd) {
     iobuf key;
-    kafka::response_writer w(key);
-    w.write(version);
+    reflection::serialize(key, version(), type(), pid.id);
```

Comment:
1. I assume the (type, id) are the semantic data being compared? it seems problematic to include the version in the key in such a way that it will be involved the comparison of two keys. What's the rationale here / plan for using the version, and making it compatible with byte-by-byte key comparison performed in the compaction layer? I could see removing the version entirely from the key (probably moving it to the value) if the key design needs to be co-designed for compaction.

2. Is this backwards compatible? If not, do we need to include any special upgrade instructions for beta users or handle the conversion for them automatically? Otherwise it would seem the behavior would be a crash.

Reply:
2. It's definitely not a beta version and we haven't announced the experimental support yet. Nobody should use it in the current state so caring about backward compatibility now is wasteful. Also it's probably gonna be changed once again when we introduce timeouts and GC. After that the format should settle, we'll make the public announcement and became backwards compatible.

Reply:
moved the version to a record's value and changed its type to int8_t